src/dstack/_internal/core/backends/cloudrift/api_client.py

```diff
@@ -0,0 +1,228 @@
+import os
+import re
+from typing import Any, Dict, List, Mapping, Optional, Union
+
+import requests
+from packaging import version
+from requests import Response
+
+from dstack._internal.core.errors import BackendError, BackendInvalidCredentialsError
+from dstack._internal.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+CLOUDRIFT_SERVER_ADDRESS = "https://api.cloudrift.ai"
+CLOUDRIFT_API_VERSION = "2025-05-29"
+
+
+class RiftClient:
+    def __init__(self, api_key: Optional[str] = None):
+        self.server_address = CLOUDRIFT_SERVER_ADDRESS
+        self.public_api_root = os.path.join(CLOUDRIFT_SERVER_ADDRESS, "api/v1")
+        self.internal_api_root = os.path.join(CLOUDRIFT_SERVER_ADDRESS, "internal")
+        self.api_key = api_key
+
+    def validate_api_key(self) -> bool:
+        """
+        Validates the API key by making a request to the server.
+        Returns True if the API key is valid, False otherwise.
+        """
+        try:
+            response = self._make_request("auth/me")
+            if isinstance(response, dict):
+                return response.get("email", False)
+            return False
+        except BackendInvalidCredentialsError:
+            return False
+        except Exception as e:
+            logger.error(f"Error validating API key: {e}")
+            return False
+
+    def get_instance_types(self) -> List[Dict]:
+        request_data = {"selector": {"ByServiceAndLocation": {"services": ["vm"]}}}
+        response_data = self._make_request("instance-types/list", request_data)
+        if isinstance(response_data, dict):
+            return response_data.get("instance_types", [])
+        return []
+
+    def list_recipies(self) -> List[Dict]:
+        request_data = {}
+        response_data = self._make_request("recipes/list", request_data)
+        if isinstance(response_data, dict):
+            return response_data.get("groups", [])
+        return []
+
+    def get_vm_recipies(self) -> List[Dict]:
+        """
+        Retrieves a list of VM recipes from the CloudRift API.
+        Returns a list of dictionaries containing recipe information.
+        """
+        recipe_group = self.list_recipies()
+        vm_recipes = []
+        for group in recipe_group:
+            tags = group.get("tags ", [])
+            has_vm = "vm" in tags
+            if group.get("name", "").lower() != "linux" and not has_vm:
+                continue
```

Comment:
This snippet doesn't seem to work as expected.

- The space in `"tags "` is unintentional.
- The `vm` tag is actually in upper case in the API.
- I assume the image is only suitable if it's both Linux and supports VMs.

```suggestion
            tags = group.get("tags", [])
            has_vm = "vm" in map(str.lower, tags)
            if group.get("name", "").lower() != "linux" or not has_vm:
                continue
```